### PR TITLE
Go to call block or alias within task block

### PR DESCRIPTION
**Pull request by ghost**
Added tests for parser, annotations and references.

### CODE REVIEW

File: src/winstanley/WdlAnnotator.scala [hunk 1]
```diff
@@ -2,7 +2,7 @@ package winstanley
 
 import com.intellij.lang.annotation.{AnnotationHolder, Annotator}
 import com.intellij.psi.PsiElement
-import winstanley.psi.{WdlCallableLookup, WdlVariableLookup}
+import winstanley.psi.{WdlCallableLookup, WdlLookup}
```
Comment (cjllanwarne) [inline src/winstanley/WdlAnnotator.scala]: I liked the differentiation that `VariableLookup` gave us... maybe something like `ValueLookup`?

File: src/winstanley/WdlAnnotator.scala [hunk 2]
```diff
@@ -12,23 +12,23 @@ class WdlAnnotator extends Annotator {
     * Makes sure that the passed PSI element is one of the cases, then determines if it's been declared somewhere
     */
   override def annotate(psiElement: PsiElement, annotationHolder: AnnotationHolder): Unit = psiElement match {
-    case value: WdlVariableLookup =>
+    case value: WdlLookup =>
 
       value.getIdentifierNode foreach { identifier =>
         val identifierText = identifier.getText
-        val declarationNames = value.findDeclarationsAvailableInScope.flatMap(_.declaredValueName)
+        val referenceNames = value.findReferencesInScope.flatMap(_.declaredValueName)
```
Comment (cjllanwarne) [inline src/winstanley/WdlAnnotator.scala]: Same here, I like valueNames

File: src/winstanley/WdlAnnotator.scala [hunk 3]
```diff
@@ -12,23 +12,23 @@ class WdlAnnotator extends Annotator {
...
+        if (!referenceNames.contains(identifierText)) {
           annotationHolder.createErrorAnnotation(identifier.getTextRange, s"No declaration found for '${identifier.getText}'")
         }
       }
     case value: WdlCallableLookup =>
 
       value.getFullyQualifiedName.getIdentifierNode foreach { identifier =>
         val identifierText = identifier.getText
-        val declarationNames = value.findTaskDeclarationsAvailableInScope.flatMap(_.declaredValueName)
+        val referenceNames = value.findTasksInScope.flatMap(_.declaredValueName)
```
Comment (cjllanwarne) [inline src/winstanley/WdlAnnotator.scala]: And here I like `taskName = value.findCallableDefinitions`

File: src/winstanley/psi/impl/WdlPsiImplUtil.scala [hunk 4]
```diff
@@ -24,7 +24,11 @@ sealed trait WdlNamedElementImplUtil {
   def getName(namedElement: WdlNamedElementImpl): String = namedElement.declaredValueName.orNull
   // TODO: Implement for "refactor/rename" functionality
   def setName(namedElement: WdlNamedElementImpl, newName: String): PsiElement = ???
-  def getNameIdentifier(namedElement: WdlNamedElementImpl): PsiElement = namedElement.getIdentifierNode.map(_.getPsi).orNull
+  def getNameIdentifier(namedElement: WdlNamedElementImpl): PsiElement = namedElement match {
+    case callable: WdlCallableLookupImpl =>
+      callable.getFullyQualifiedName.getLastChild
```
Comment (cjllanwarne) [inline src/winstanley/psi/impl/WdlPsiImplUtil.scala]: ToL: might look neater if you combine these two lines into one

File: src/winstanley/references/WdlReference.scala [hunk 5]
```diff
@@ -4,14 +4,14 @@ import javax.annotation.Nullable
...
 import com.intellij.openapi.util.TextRange
 import com.intellij.psi.{PsiElement, PsiReferenceBase}
-import winstanley.psi.WdlVariableLookup
+import winstanley.psi.{WdlLookup, WdlNamedElement}
 import winstanley.structure.WdlImplicits._
 
 /**
-  * Holds a PSI element that references a variable. Defines methods to find and return the named
-  * element for the variable being referenced.
+  * Holds a PSI element that is referenced by a WdlLookup. Defines methods to find and return the named
```
Comment (cjllanwarne) [inline src/winstanley/references/WdlReference.scala]: `"... that might be referenced by..."`

**Review by cjllanwarne — COMMENTED** [review]
A couple more test suggestions:

```
call x as y
# Should be an error annotation:
Int z = x.out 
```

```
call x
call x as y
Int a = x.out
# Check this `y.out` reference goes to the right call
Int b = y.out
```

File: src/winstanley/WdlAnnotator.scala (hunk 1 above)
Comment (ghost) [reply src/winstanley/WdlAnnotator.scala]: Done

File: src/winstanley/WdlAnnotator.scala (hunk 3 above)
Comment (ghost) [reply src/winstanley/WdlAnnotator.scala]: Done

File: src/winstanley/WdlAnnotator.scala (hunk 2 above)
Comment (ghost) [reply src/winstanley/WdlAnnotator.scala]: Done

**Comment by cjllanwarne** [comment]
Awesome stuff. 👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/winstanley/pull-request/62/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/winstanley)

**Pull request merged**